warehouse/accounts/services.py

```diff
@@ -643,6 +643,11 @@ def loads(self, token, return_timestamp=False):
 
         return data
 
+    def unsafe_load_payload(self, token):
+        _, data = self.serializer.loads_unsafe(token)
+
+        return data
```

Comment:
It is used when reconstituting the expired token from DB, since that's the only place the "desired_role" is stored: https://github.com/pypi/warehouse/pull/13164/files#diff-55d42e8160b62924651b14b5c3649c6511c24a79876c8447cdad028da727d14fR2169

I will update with a comment to discourage use though!

Reply:
Also, just noted that we can at least ensure the signature was valid along the way, adding that to fixup.

see d1581702cbcd21f50663eab23a41882ef624886b